plugins/module_utils/encode.py

```diff
@@ -194,17 +195,15 @@ def temp_data_set(self, reclen, space_u):
         if self.tmphlq:
             hlq = self.tmphlq
         else:
-            hlq = datasets.hlq()
-        temp_ps = datasets.tmp_name(hlq)
-        response = datasets._create(
+            hlq = datasets.get_hlq()
+        temp_ps = datasets.tmp_name(high_level_qualifier=hlq)
+        datasets.create(
             name=temp_ps,
             type="SEQ",
             primary_space=size,
             record_format="VB",
             record_length=reclen,
         )
-        if response.rc:
```

Comment:
This is because I removed the internal function 
```
response = datasets._create(
                        name=temp_ps,
                        type="SEQ",
                        primary_space=size,
                        record_format="VB",
                        record_length=reclen,
                    )
```
that returned a ZOAUResponse object to

```
        datasets.create(
            name=temp_ps,
            type="SEQ",
            primary_space=size,
            record_format="VB",
            record_length=reclen,
        )
```

which according to ZOAU documentation will return a DataSet object , I just changed this in the code to get the data set object and actually return the data set name in the data set object created by ZOAU. My logic on removing the OSError, is to instead of having to catch ZOAUException or DatasetVerificationError and mask it into a OSError with a vague error message like "Failed when allocating temporary sequential data set!", rather raise the specific ZOAU error with better details on failure.

Reply:
Good question and good answer, I like this approach :), this will work, interesting enough is ZOAUException is raised by ZOAU when RC >=8  :). 

The error was vague so this approach is better; our modules are regularly creating stateful objects (temporary files and data sets) unbeknown to the user so even floating an error up from ZOAU might be a bit confusing not much we can do but possibly make a general statement in every module notes section that we might create temporary objects (not this PR). 

One suggestion though, if we don't throw the OSError anymore, could we remove it?
```
Raises:
            OSError: When any exception is raised during the data set allocation
``` 

What are your thoughts about adding `ZOAUException` or `DatasetVerificationError` to our `Raises` instead so our team has easy access or is this to dynamic? The [style guide](https://numpydoc.readthedocs.io/en/latest/format.html#raises) says:

> This section should be used judiciously, i.e., only for errors that are non-obvious or have a large chance of getting raised.

One other thing, and I can do this is to post in our internal channel for those using our module_utils of this minor change, just to let them know.